inspectit-oce-core/src/main/java/rocks/inspectit/oce/core/rocks/inspectit/oce/core/config/DirectoryPropertySource.java

```diff
@@ -0,0 +1,146 @@
+public class DirectoryPropertySource extends EnumerablePropertySource<Void> {
+
+
+    private static final List<String> PROPERTIES_ENDINGS = Arrays.asList(".properties");
+    private static final List<String> YAML_ENDINGS = Arrays.asList(".yml", ".yaml");
+
+    @Getter
+    private Path rootDir;
+
+    /**
+     * stores all contained PropertySources (=one for each contained file)
+     */
+    private final List<PropertySource<?>> ownedFilePropertySources = new ArrayList<>();
+
+    private MutablePropertySources container;
+
+    /**
+     * Factory method.
+     * Creates a DirectoryPropertySource and places it as last source in the given container.
+     * @param name the name of the DirectoryPropertySource to create. This is used as basename for the PropertySources generated for each contained file.
+     * @param rootDir the directory to parse for config files
+     * @param container the container to append the PropertySources to
+     * @return
+     */
+    public static DirectoryPropertySource addLast(String name, Path rootDir, MutablePropertySources container) {
+        val dps = new DirectoryPropertySource(name,rootDir,container);
+        container.addLast(dps);
+        dps.reload();
+        return dps;
+    }
+
+    private DirectoryPropertySource(String name, Path rootDir, MutablePropertySources container) {
+        super(name);
+        this.rootDir = rootDir;
+        this.container = container;
+    }
+
+    /**
+     * Removes all previously loaded {@link PropertySource}s generated by this {@link DirectoryPropertySource} .
+     * Afterwards parses the directory and generated a {@link PropertySource} for each found config file, which is added after this {@link DirectoryPropertySource} to the container.
+     */
+    public void reload() {
+        for(PropertySource<?> ps : ownedFilePropertySources) {
+            container.remove(ps.getName());
+        }
+        ownedFilePropertySources.clear();
+
+        if (loadContentsToPropertySources()) return;
+
+        PropertySource<?> previous = this;
+        for(var pps : ownedFilePropertySources) {
+            container.addAfter(previous.getName(), pps);
+            previous = pps;
+        }
+    }
+
+    private boolean loadContentsToPropertySources() {
+        Stream<Path> files;
+        try {
+            files = Files.list(rootDir).filter(p -> !p.toFile().isDirectory());
+        } catch (IOException e) {
+            logger.error("Unable to access config dir", e);
+            return true;
+        }
+        //alphabetical order, last loaded file wins
+        files.sorted(Comparator.naturalOrder()).forEachOrdered(file -> {
+            if (doesFileHaveEnding(file, PROPERTIES_ENDINGS)) {
+                loadPropertiesFile(file);
+            } else if (doesFileHaveEnding(file, YAML_ENDINGS)) {
+                loadYamlFile(file);
+            }
+        });
+        return false;
+    }
+
+    private void loadPropertiesFile(Path file) {
+        try {
```

Comment:
How about if we use the same code as for `loadYamlFile`, you just need another method in the `ConfigurationUtils` and use `PropertiesPropertiesFactoryBean` instead of `YamlPropertiesFactoryBean` ?